### DISCUSSION TITLE
Performance improvment

### DISCUSSION THREAD

**Discussion by flaxy123** [199ocero/filachat#9] [Ideas]
Hi, first I want to say that Filachat is an impressive project with a lot of potential. The overall functionality is great, and it integrates well into my Laravel and Filament-based application. However, I have noticed a few key areas that could benefit from improvements, especially around performance and usability.

At the moment, Filachat is quite slow to load and respond. Every time I open a chat or initiate a new conversation, there’s a noticeable delay. This lag affects the user experience significantly, especially when dealing with multiple conversations. Improving the speed and overall responsiveness would make Filachat much more efficient for real-time communication.

Feature Suggestions

Group Messaging Support:
Adding group messages would be an incredibly valuable feature. Being able to communicate in group chats is essential for many team-based applications and would make Filachat much more versatile.

Direct User Search in Input Field Instead of Modals:
Currently, using modals for user search and conversation creation slows down the process. Having a direct input field where users can search for and select contacts would streamline this action and improve the overall speed of the interface.

Optimize for Faster Load Times:
Consider optimizing the queries and the general handling of conversation lists, messages, and other data. As the app grows with more users and messages, it becomes crucial to ensure the application can scale efficiently.

**Comment by 199ocero** [comment]
Thank you so much for pointing out any issues with the plugin @flaxy123  this is on todo list and also your feature suggestions. This plugin is my side project and will only be updated when I have free time so I'm not really sure if these features and issues will be solve as early as possible BUT any PR fix this will be greatly appreciated. Thank you!
